webapp/src/test/java/rocks/metaldetector/service/home/HomepageServiceImplTest.java

```diff
@@ -0,0 +1,126 @@
+    underTest.createHomeResponse();
+
+    // then
+    verify(followArtistService, times(1)).getFollowedArtistsOfCurrentUser();
+  }
+
+  @Test
+  @DisplayName("releaseService is called with followed artists' names")
+  void test_release_service_called_with_followed_artists_names() {
+    // given
+    var followedArtists = List.of(ArtistDtoFactory.withName("A"), ArtistDtoFactory.withName("B"));
+    var followedArtistsNames = List.of("A", "B");
+    doReturn(followedArtists).when(followArtistService).getFollowedArtistsOfCurrentUser();
+
+    // when
+    underTest.createHomeResponse();
+
+    // then
+    verify(releaseService, times(1)).findReleases(eq(followedArtistsNames), any(), any());
+  }
+
+  @Test
+  @DisplayName("releaseService is called with correct dates")
+  void test_release_service_called_with_correct_dates() {
+    // given
+    var expectedDate = LocalDate.now();
+    doReturn(List.of(ArtistDtoFactory.withName("A"), ArtistDtoFactory.withName("B")))
+        .when(followArtistService).getFollowedArtistsOfCurrentUser();
+
+    // when
+    underTest.createHomeResponse();
+
+    // then
+    verify(releaseService, times(1)).findReleases(any(), eq(expectedDate), eq(expectedDate.plusMonths(1)));
+  }
+
+  @Test
+  @DisplayName("releaseService is not called with user does not follow any artists")
+  void test_release_service_not_called() {
+    // given
+    doReturn(Collections.emptyList()).when(followArtistService).getFollowedArtistsOfCurrentUser();
+
+    // when
+    underTest.createHomeResponse();
+
+    // then
+    verifyNoInteractions(releaseService);
+  }
+
+  @Test
+  @DisplayName("upcoming releases are returned")
+  void test_upcoming_releases_are_returned() {
+    // given
+    List<ReleaseDto> upcomingReleases = List.of(ReleaseDtoFactory.withArtistName("A"), ReleaseDtoFactory.withArtistName("B"));
+    doReturn(List.of(ArtistDtoFactory.createDefault())).when(followArtistService).getFollowedArtistsOfCurrentUser();
+    doReturn(upcomingReleases).when(releaseService).findReleases(any(), any(), any());
+
+    // when
+    var result = underTest.createHomeResponse();
+
+    // then
+    assertThat(result.getUpcomingReleases()).isEqualTo(upcomingReleases);
+  }
+
+  @Test
+  @DisplayName("only the first four upcoming releases are returned")
+  void test_upcoming_releases_limit() {
+    // given
+    var tooMuch = ReleaseDtoFactory.withArtistName("E");
+    var upcomingReleases = List.of(
+        ReleaseDtoFactory.withArtistName("A"), ReleaseDtoFactory.withArtistName("B"),
+        ReleaseDtoFactory.withArtistName("C"), ReleaseDtoFactory.withArtistName("D"),
+        tooMuch);
+    doReturn(List.of(ArtistDtoFactory.createDefault())).when(followArtistService).getFollowedArtistsOfCurrentUser();
+    doReturn(upcomingReleases).when(releaseService).findReleases(any(), any(), any());
+
+    var result = underTest.createHomeResponse();
+
+    // then
+    assertThat(result.getUpcomingReleases().size()).isEqualTo(RESULT_LIMIT);
```

Comment:
```suggestion
    assertThat(result.getUpcomingReleases().size()).isEqualTo(4);
```

Otherwise the test description will at some point no longer correspond to reality.